chain/default_syncer.go

```diff
@@ -255,6 +255,9 @@ func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.Tip
 	}
 
 	if heavier {
+		if syncer.IsReorg(ancestors, next) {
```

Comment:
this call assumes the ancestors list is sufficiently long to reach the current head